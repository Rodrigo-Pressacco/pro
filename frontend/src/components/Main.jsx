import React, { createContext, useState } from "react";
import Listado from "./Listado";
import Buscador from "./Buscador"
import Categorias from "./Categorias";
import { BrowserRouter, Route, Switch } from 'react-router-dom'
import Header from "./Header";
import Registro from "./Registro";
import Login from "./Login";
import Detalle from "./Detalle";
import Reserva from "./Reserva";
import fechasContext from "../contexts/fechasContext";


function Main() {

    /*states para header*/
    const [headerIniciarSesion, setHeaderIniciarSesion] = useState(false);
    const [headerRegistrarse, setHeaderRegistrarse] = useState(false);
    const [headerLogeado, setHeaderLogeado] = useState(false);
    const [datosUsuario, setDatosUsuario] = useState([{
        nombre: "hola",
        apellido: "hola"
    }, false]);

    /*states para filtros*/
    const [filtCat, setFiltCat] = useState(false)
    const [filtCity, setFiltCity] = useState(false)
    const [filterCategoria, setFilterCategoria] = useState();
    const [filterDestino, setFilterDestino] = useState();
    const [listCategoria, setListCategoria] = useState();
    const [listDestino, setListDestino] = useState();

    /*states para card*/
    const [dataCard, setDataCard] = useState(["",false])

    /*State para detalle para ir a reserva*/
    const [estoyLoguado, setEstoyLoguado] = useState(false)

    // States para Context de Fechas
    const [fechaInicial,setFechaInicial] = useState(new Date())
    const [fechaFinal,setFechaFinal] = useState(new Date())

    const handleClickChange = (datos) => {
        if (datos === 1) {
            setHeaderIniciarSesion(false)
            setHeaderRegistrarse(false)
        }
        if (datos === 2) {
            setHeaderRegistrarse(true)
            setHeaderIniciarSesion(false)
        }
        if (datos === 3) {
            setHeaderIniciarSesion(true)
            setHeaderRegistrarse(false)
        }
    }

    const handleSelected = (datos) => {
        setHeaderLogeado(true)

    };

    const handleDestinos = (name) => {
        setFilterDestino(name)
        setFiltCat(false)

    }

    const handleCategorias = (name) => {
        setFilterCategoria(name)
    }
    const handleListCategoria = (list) => {
        setListCategoria(list)
        setFiltCat(true)
    }
    const handleListDestino = (list) => {
        setListDestino(list)
        setFiltCity(true)
    }

    const handleRemoveFilter = () => {
        setFiltCat(false)
        setFiltCity(false)
    }

    const handleData = (data) =>{
        setDataCard([data , true]);
    }

    const handleLogeado = (data) =>{
        setEstoyLoguado(data)
    }

    const handleDatosUsuario = (data) =>{
        setDatosUsuario([data,true])
    }

    return (

        <div className="body">
                <BrowserRouter >
                    <Header header={{ headerIniciarSesion, headerRegistrarse, headerLogeado, datosUsuario}} cambiarBotton={handleClickChange} />
                    <Switch>
                        <Route exact path="/" render={() =>
                            <>
                                <Buscador onSelect={handleDestinos} onSelect2={handleListDestino} />
                                <Categorias onSelect={handleCategorias} onSelect2={handleListCategoria} />
                                <Listado filterCategory={filterCategoria} filterLocation={filterDestino}
                                        filterCategoryList={[listCategoria, filtCat]} filterLocationList={[listDestino, filtCity]}
                                        onFilter={handleRemoveFilter} handleDataCard={handleData} />
                            </>
                        } />
                        <Route exact path="/login" render={() =>
                            <Login cambiarBotton={handleClickChange} onSelect={handleSelected} logeado={handleLogeado} dataUsuario={handleDatosUsuario} />
                        } />
                        <Route exact path="/registrarme" render={() =>
                            <Registro cambiarBotton={handleClickChange} />
                        } />
                            <Route exact path="/detalle/:id" render={() =>
                                <fechasContext.Provider value={{fechaInicial,setFechaInicial,fechaFinal,setFechaFinal}}>
                                    < Detalle datos={dataCard} />
                                </fechasContext.Provider>
                            } />
                            <Route path="/detalle/:id/reserva" render={() =>
                                <fechasContext.Provider value={{fechaInicial,setFechaInicial,fechaFinal,setFechaFinal}}>
                                    < Reserva />
                                </fechasContext.Provider>
                            } />
                    </Switch>

                </BrowserRouter>
        </div>
    )
}
export default Main;