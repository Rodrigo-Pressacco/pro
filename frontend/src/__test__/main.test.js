import React from "react";
import { shallow } from "enzyme";
import "@testing-library/jest-dom";
import Main from '../components/Main'

import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import { createSerializer } from 'enzyme-to-json';
Enzyme.configure({ adapter: new Adapter() });
expect.addSnapshotSerializer(createSerializer({ mode: 'deep' }));

describe('<Main />', () => {
    let main
    let handleClickChange = jest.fn(); 
    let handleSelected = jest.fn();
    let handleDestinos = jest.fn();
    let handleCategorias = jest.fn();
    let handleListCategoria = jest.fn();
    let handleListDestino = jest.fn();
    let handleRemoveFilter = jest.fn();
    let handleData = jest.fn();
    let handleLogeado = jest.fn();
    let handleDatosUsuario = jest.fn();

    
    beforeEach(() => {
        main = shallow(<Main    handleClickChange={handleClickChange}
                                handleSelected={handleSelected}
                                handleDestinos={handleDestinos}
                                handleCategorias={handleCategorias}
                                handleListCategoria={handleListCategoria}
                                handleListDestino={handleListDestino}
                                handleRemoveFilter={handleRemoveFilter}
                                handleData={handleData}
                                handleLogeado={handleLogeado}
                                handleDatosUsuario={handleDatosUsuario}
                                 />)
    });

    it("Deberia mostrar <Main/> correctamente", () => {
        expect(main).toMatchSnapshot();
    });

})