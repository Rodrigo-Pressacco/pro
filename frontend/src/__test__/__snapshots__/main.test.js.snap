// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> Deberia mostrar <Main/> correctamente 1`] = `
<div
  className="body"
>
  <BrowserRouter>
    <Header
      cambiarBotton={[Function]}
      header={
        Object {
          "datosUsuario": Array [
            Object {
              "apellido": "hola",
              "nombre": "hola",
            },
            false,
          ],
          "headerIniciarSesion": false,
          "headerLogeado": false,
          "headerRegistrarse": false,
        }
      }
    />
    <Switch>
      <Route
        exact={true}
        path="/"
        render={[Function]}
      />
      <Route
        exact={true}
        path="/login"
        render={[Function]}
      />
      <Route
        exact={true}
        path="/registrarme"
        render={[Function]}
      />
      <Route
        exact={true}
        path="/detalle/:id"
        render={[Function]}
      />
      <Route
        path="/detalle/:id/reserva"
        render={[Function]}
      />
    </Switch>
  </BrowserRouter>
</div>
`;
